parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -102,6 +103,13 @@
 
   public static String PARQUET_READ_PARALLELISM = "parquet.metadata.read.parallelism";
 
+  // configure if we want to use Hadoop's V2 read(bytebuffer) API.
+  // If true, we try to read using the new Hadoop read(ByteBuffer) api. This reads data into the provided
+  // byteBuffer and allows us to potentially take advantage zero-copy read path in Hadoop.
+  // Else, we skip and either read into the byteBuffer's array (if its present) or allocate one and copy
+  // into it.
+  public static String PARQUET_HADOOP_BYTEBUFFER_READ = "parquet.read.use.byte.buffer";
```

Comment:
Yeah so @isnotinvain and I had some discussion about this property as well (see his comments above). So his suggestion was to get rid of this property all together. We can either:
1) Cache in a static var whether we computed Hadoop V1 / V2 classes and reuse that. Saves us the reflection call on subsequent files. 
2) Perform the Hadoop V1 / V2 check every time we start reading a file. If there is a chance that the Hadoop V1 / V2 situation changes in the lifetime of the JVM, this will help us pick up the appropriate approach. Not sure what the use-case is for this though. I added the knob here to save us the reflection call when you know for sure you're on Hadoop V1. 

Personally I like option 1. We pay the cost once and it helps us get rid of one knob to tune. Thoughts?